tools/sigclient/pkg/ingest/ingest.go

```diff
@@ -173,18 +225,24 @@ func runIngestion(iType IngestType, rdr utils.Generator, wg *sync.WaitGroup, url
 
 	i := 0
 	var bb *bytebufferpool.ByteBuffer
+	var payload []byte
+	var err error
 	maxDuration := 2 * time.Hour
-	for continous || eventCounter < totalEvents {
+	for continuous || eventCounter < totalEvents {
 
 		recsInBatch := batchSize
-		if !continous && eventCounter+batchSize > totalEvents {
+		if !continuous && eventCounter+batchSize > totalEvents {
 			recsInBatch = totalEvents - eventCounter
 		}
 		i++
 		if iType == ESBulk {
 			bb = bytebufferpool.Get()
 		}
-		payload, err := generateBody(iType, recsInBatch, i, rdr, actLines, bb)
+		if iType == OpenTSDB {
+			payload, err = generateBodyFromPredefinedSeries(recsInBatch, &seriesId)
```

Comment:
you dont need to pass `seriesID` as a var, just add the `recsInBatch` to it after this func call. It also saves the additions you do everytime in the for loop in that function.